addons/website_event_sale/controllers/main.py

```diff
@@ -49,6 +49,7 @@ def _create_attendees_from_registration_post(self, event, registration_data):
                 data['sale_order_line_id'] = cart_data[event_ticket_id]
 
         request.session['website_sale_cart_quantity'] = order_sudo.cart_quantity
+        request.session['sale_last_order_id'] = order_sudo.id
```

Comment:
Hello @pedrambiria 
Got a look and imo it would be better if we do this in `registration_confirm()` just before the
`return request.redirect("/shop/checkout")` as we don't care about the order if registrations are free.
Cheers!

Reply:
Hello @jeh-odoo 
Thanks for your review.
I modified the code.